DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/board_test/src/board_test.cpp

```diff
@@ -203,6 +203,18 @@ int main(int argc, char* argv[]) {
 
       ret |= hldshim.KernelMemBW(q);
     }
+
+    // Test 7 - USM
+    if (test_to_run == 0 || test_to_run == 7) {
+      std::cout << "\n*****************************************************************\n"
+                << "***********************  USM Bandwidth  *************************\n"
+                << "*****************************************************************\n\n";
+#if defined(SUPPORTS_USM)
+      ret |= hldshim.USMBWTest(q);
+#else
+      std::cout << "Board does not support USM, skipping this test \n";
+#endif
```

Comment:
we can detect this at runtime:

```
    // make sure the device supports USM host allocations
    if (!device.has(sycl::aspect::usm_host_allocations)) {
      std::cerr << "This design must either target a board that supports USM "
                   "Host/Shared allocations, or IP Component Authoring. "
                << std::endl;
      std::terminate();
    }
```

you can add this snippet to line 72 of `board_test.cpp`. see an example of it in fpga_template https://github.com/oneapi-src/oneAPI-samples/blob/master/DirectProgramming/C%2B%2BSYCL_FPGA/Tutorials/GettingStarted/fpga_template/src/fpga_template.cpp

Reply:
Same as comment above.